### PR TITLE
MSI-1921: Replaced inventory_stock_1 view for product page

### PR DESCRIPTION
**Pull request by Stepa4man**
### Description
MSI-1921: Replaced inventory_stock_1 with cataloginventory_stock_status until bundle product is not implemented

### Fixed Issues
1. magento/magento2#1921: Performance degradation on product page

### Manual testing scenarios
Described in issue

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php [hunk 1]
```diff
@@ -27,37 +28,59 @@ class GetStockItemData implements GetStockItemDataInterface
...
     /**
      * @param ResourceConnection $resource
      * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+     * @param DefaultStockProviderInterface $defaultStockProvider
      */
     public function __construct(
         ResourceConnection $resource,
-        StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        DefaultStockProviderInterface $defaultStockProvider
```
Comment (nuzil) [inline app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php:44]: Basically if MSI is support backward compatibility, this part is incompatible, as we cannot add new params in construct))

File: app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php [hunk 2]
```diff
@@ -27,37 +28,59 @@ class GetStockItemData implements GetStockItemDataInterface
...
     /**
      * @inheritdoc
      */
     public function execute(string $sku, int $stockId): ?array
     {
-        $stockItemTableName = $this->stockIndexTableNameResolver->execute($stockId);
+        $isDefaultStock = $stockId === $this->defaultStockProvider->getId();
+        if ($isDefaultStock) {
+            $stockItemTableName = $this->resource->getTableName('cataloginventory_stock_status');
+            $qtyColumnName = 'qty';
```
Comment (nuzil) [inline app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php:59]: Can we also use constant here?

File: app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php [hunk 3]
```diff
@@ -27,37 +28,59 @@ class GetStockItemData implements GetStockItemDataInterface
...
+            $isSalableColumnName = 'stock_status';
+        } else {
```
Comment (nuzil) [inline app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php:61]: Can we optimise those if/else statement? For example define in advance and move them away from conditions and change only if required
$qtyColumnName = IndexStructure::QUANTITY;
$isSalableColumnName = IndexStructure::IS_SALABLE;

**Review by nuzil — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php (hunk 2 above)
Comment (maghamed) [reply app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php:59]: @nuzil private constant - yes

**Comment by maghamed** [comment]
hey @Stepa4man  does this PR is ready for review? 

If so, could you please unassigned "WIP" (work in progress) label, as currently it prevents us from  reviewing it, as it's not clear whether you finished or not

**Pull request closed without merging**